### PR TITLE
remove-w2ui #9: Statusbar

### PR DESCRIPTION
**Pull request by eszkadev**
Statusbar reworked to JSDialog.

New:
statusbar

Old:
statusbarold

Later we will need to adjust styling as it is different (darker separators, update icons).

### CODE REVIEW

File: browser/src/control/Control.StatusBar.js [hunk 1]
```diff
@@ -74,195 +73,78 @@
...
+			}, 800);
+		}
+	},
 
-		this.map.on('zoomend', function () {
-			var zoomPercent = 100;
-			var zoomSelected = null;
-			switch (that.map.getZoom()) {
+	onZoomEnd: function() {
+		var zoomPercent = 100;
```
[the remark this answers is not in the document]
Comment (eszkadev) [reply browser/src/control/Control.StatusBar.js:144]: false positive

File: browser/src/control/Control.StatusBar.js [hunk 2]
```diff
@@ -507,45 +470,43 @@
...
+			} else {
 				state = this.toLocalePattern('Page %1 of %2', 'Slide (\\d+) of (\\d+)', state, '%1', '%2');
-			this.updateToolbarItem(statusbar, 'PageStatus', $('#PageStatus').html(state ? state : '&nbsp;&nbsp;&nbsp;&nbsp;&nbsp').parent().html());
+				this.updateHtmlItem('PageStatus', state ? state : '&nbsp;&nbsp;&nbsp;&nbsp;&nbsp');
+			}
 		}
 	},
 
 	onLanguagesUpdated: function() {
+		var menuEntries = JSDialog.MenuDefinitions.get('LanguageStatusMenu');
```
[the remark this answers is not in the document]
Comment (eszkadev) [reply browser/src/control/Control.StatusBar.js]: done

**Review by gokaysatir — COMMENTED** [review]
Thanks for this big step forward :) I have some notes below.

Desktop:
    * Writer + Calc: There is a file-not found error: /browser/4c70d77f31/images/lc_zoom-menu.svg
    * Writer + Calc + Impress ReadOnly: Set language option is visible in readonly view mode. And also usable for some time.
    * Writer + Impress: Previous and next page buttons are enabled when there is only one page. They were used to get disabled when there is only one page or up was disabled when the first page is selected and down was disabled when the last page is selected.
    * Calc: Search next - previous buttons are enabled before the search happens.
    * Calc: File not found error: /browser/4c70d77f31/images/lc_statetablecellmenu-menu.svg
    * Calc readonly: StateTableCellMenu is enabled but doesn't work. We should possibly hide it.
    * All: Edit - Readonly label looks and behaves like a button. It'd be better to make it look like a label. It was centered before. Now it is on the left.
    
Writer Mobile:
    * Comment button is not visible on top toolbar.
    * Comments are not centered in the wizard.
    * The alignment button looks like it's disabled on the bottom toolbar. Same for other dropdown menus. Arrow colors are light.
    * Next and previous search buttons are not visible.
    * Couldn't make text color picker work.

Calc Mobile:
    * Insert rows below doesn't have an icon.
    * Insert columns after doesn't have an icon.
    * Search: Previous and next buttons are not visible.

Impress Mobile:
    * Selected slide rectangle is not centered.
    * ReadOnly: Context menu should be disabled.
    * Search: Next and previous buttons are not visible.
    * ReadOnly: There is no search button.
    * Edit view: Context menu opens downwards and not usable.

**Review by gokaysatir — APPROVED** [review]
Thanks @eszkadev :)

**Comment by eszkadev** [comment]
> Thanks for this big step forward :) I have some notes below.
> 
> Desktop: * Writer + Calc: 
...
I will fix desktop findings. Thanks for testing.

> Writer Mobile:
> Calc Mobile:
> Impress Mobile:

These shouldn't be impacted as it reworked "statusbar" not mobile toolbars.

**Pull request merged**